### DISCUSSION TITLE
Added 'Use relative size' function to all skin HUD Components

### DISCUSSION THREAD

**Discussion by huyenden** [ppy/osu#25610] [Ideas] [answered]
I have a skin arranged like this:

https://github.com/ppy/osu/assets/128424762/3d8ce0dc-dae8-4ade-a473-2f0766af4afa

When I shared the skin with my friends, some people said that my skin was broken. Then I discovered it was because the other person's UI scale was not 1.00x
![osu! 2023-11-30 19-20-47-283](https://github.com/ppy/osu/assets/128424762/459761d9-ec6d-4a91-b31e-62790f47b9c2)

So I think if I add the Use relative size of the HP argon bar function to the all HUD options, it will help the UIs not break when changing the UI scale like the skin editor wants.
![osu! 2023-11-30 19-26-24-766](https://github.com/ppy/osu/assets/128424762/42c1cb8a-42a2-473d-aa4f-78f743ef03c3)

Is this possible? Please let me know what you think <3

**Comment by Saraistupid** [comment]
why doesn't this have any replies? 😭 any HUD editor should have relative scaling

**Reply by huyenden** [reply]
For now I see only a few like the HP Argon bar that scale to this, I predict it will be a long time.

**Comment by vatei** [comment]
Sounds super useful and needed yeah

**Comment by peppy** [comment, marked as answer]
I think we'd sooner just disable ui scale affecting gameplay elements.

**Reply by huyenden** [reply]
Of course I would expect that, UI Scale just resizing the settings interface and map selection interface would be fine

**Reply by Saraistupid** [reply]
but why? wouldn't relative size scaling be a better option? people can play with different resolutions or want to make skins that fit all res which lazer hud stuff break with (at least from my experience)

i depend on ui scaling for gameplay too cuz taiko can be too small for me sometimes and changing scale without changing res is more convenient to do

**Comment by yeahirai** [comment]
Has been responded already but i would like to add another situation where not having this breaks stuff.

Many skins uses the hp bar to create custom playfields. And all of them rely on the skin element filling the whole screen vertically and horizontally.
![osu_2024-12-21_14-15-08](https://github.com/user-attachments/assets/508bfdd4-ab97-4408-b46d-ed5b5f8a9fd8)
When changing the ui scaling every single skin that changes the playfield gets broken. On the example above i'm using 0,85x scalling.
![osu_2024-12-21_14-20-45](https://github.com/user-attachments/assets/cf0f31b2-7221-4c28-8f1b-12e52faf093a)
Is this one its 1.15x
